sycl/test/basic_tests/vectors/vectors.cpp

```diff
@@ -55,6 +55,25 @@ template <typename From> void check_convert_from() {
   check_signed_unsigned_convert_to<From, double>();
 }
 
+template <typename T, typename OpT>
+void check_ops(OpT op, T c1, T c2) {
+  auto check = [&](sycl::vec<T,2> vres) {
+    assert(op(c1, c2) == vres[0]);
+    assert(op(c1, c2) == vres[1]);
+  };
+
+  sycl::vec<T, 2> v1(c1);
+  sycl::vec<T, 2> v2(c2);
+  check(op(v1.template swizzle<0,1>(), v2.template swizzle<0,1>()));
+  check(op(v1.template swizzle<0,1>(), v2));
+  check(op(v1.template swizzle<0,1>(), c2));
+  check(op(c1, v2.template swizzle<0,1>()));
+  check(op(c1, v2));
+  check(op(v1, v2.template swizzle<0,1>()));
+  check(op(v1, v2));
+  check(op(v1, c2));
+}
```

Comment:
Not familiar with this area, but should we add a test with a swizzle other than identity?